src/openrct2/core/StringBuilder.hpp

```diff
@@ -45,10 +42,9 @@ class StringBuilder final
     void Append(codepoint_t codepoint)
     {
         size_t codepointLength = String::GetCodepointLength(codepoint);
-        EnsureCapacity(_length + codepointLength + 1);
-        String::WriteCodepoint(_buffer + _length, codepoint);
-        _length += codepointLength;
-        _buffer[_length] = 0;
+        std::vector<utf8> data(codepointLength);
```

Comment:
Shouldn't this one be `std::basic_string`?

Reply:
It could be, but it is not mandatory:

The thing is that `String::WriteCodepoint` implementation will assign utf8 values for data()[0], ..., data()[3], so the "string" semantic is not needed here.

If you think it is more intuitive to use the basic_string, I can make the change.

Reply:
Yeah, I think it's better, because it conveys the meaning of "parts of a whole" better than vector, does it not? Vector kind of tells that the elements are correlated, but maybe not really connected to each other.